### PR TITLE
Add benchmarks for Zip entry extraction and creation

### PR DESCRIPTION
**Pull request by Jozkee**
cc @adamsitnik @carlossanlop

### CODE REVIEW

File: src/benchmarks/micro/libraries/System.IO.Compression/Zip.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            string currentTestFolderLocation = Path.Combine(s_testFoldersLocation, TestFolder);
+            _archiveLocation = currentTestFolderLocation + "zip";
+
+            // Create a zip archive, then open it.
+            // We will measure how much it takes to extract an entry using ExtractToFile.
+            ZipFile.CreateFromDirectory(currentTestFolderLocation, _archiveLocation, CompressionLevel, includeBaseDirectory: false);
+            _archive = ZipFile.OpenRead(_archiveLocation);
+            _entry = _archive.Entries[0];
+            _tempDestinationFileName = FileUtils.GetTestFilePath();
```
Comment (Jozkee) [inline src/benchmarks/micro/libraries/System.IO.Compression/Zip.cs:46]: Please let me know if this is the right way of getting a temporarily file name for concurrent writes/deletes. In long runs I was always getting "access to the path is denied errors", by long runs I mean this command:
```
dotnet run -c Release -f net6.0 
--filter *ExtractEntryToFile* --statisticalTest 3ms 
--coreRun 
  "C:\repos\runtime2\artifacts\bin\testhost\net6.0-windows-Release-x64\shared\Microsoft.NETCore.App\6.0.0\CoreRun.exe" 
  "C:\repos\runtime\artifacts\bin\testhost\net6.0-windows-Release-x64\shared\Microsoft.NETCore.App\6.0.0\CoreRun.exe" 
--maxIterationCount 101 --minIterationCount 100
```

Comment (carlossanlop) [reply src/benchmarks/micro/libraries/System.IO.Compression/Zip.cs:46]: @adamsitnik there was another argument you used, but I can't find it. Do you remember what was it? Should David use it here too?

**Comment by danmoseley** [comment]
@Jozkee @carlossanlop what is the status of this PR?

**Comment by danmoseley** [comment]
@carlossanlop @Jozkee is this dead or alive 😸

File: src/benchmarks/micro/libraries/System.IO.Compression/Zip.cs (hunk 1 above)
Comment (carlossanlop) [reply src/benchmarks/micro/libraries/System.IO.Compression/Zip.cs:46]: ping @adamsitnik this was so long ago that I don't remember the context of the argument I mentioned 😄

**Review by carlossanlop — APPROVED** [review]
The code looks good to me, modulo the question about how to properly open the file to ensure we don't see the error. Help us @adamsitnik, you're our only hope.

**Pull request closed without merging**